### PR TITLE
nearly complete with find by author name in iter5

### PR DESCRIPTION
**Pull request by colehart**

### CODE REVIEW

File: lib/library.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Library class
+class Library
+  attr_reader :books
+
+  def initialize
+    @books = []
+  end
+
+  def add_to_collection(book)
+    @books << book.shift
```
Comment (BrianZanti) [inline lib/library.rb]: Instead of doing this, your add_book method in author should return a book object, not an array

File: lib/author.rb [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+    @first_name = author_hash[:first_name]
+    @last_name  = author_hash[:last_name]
+    @books      = []
+  end
+
+  def add_book(title, publication_date)
+    @books << Book.new(author_first_name: @first_name,
+                       author_last_name: @last_name,
+                       title: title,
+                       publication_date: publication_date)
```
Comment (BrianZanti) [inline lib/author.rb]: This should return the new Book, not the entire @books array

File: lib/library.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  def include?(title)
+    @books.map(&:title).include?(title)
+  end
+
+  def card_catalogue
+    @books.sort_by!(&:author_last_name)
```
Comment (BrianZanti) [inline lib/library.rb]: You don't want to use the ! here because this will change your array of books

File: lib/library.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  def find_by_author(author_name)
+    name_ary = author_name.split(' ')
+    book_ary = @books.map do |book|
```
Comment (BrianZanti) [inline lib/library.rb]: don't use data type in variable names

File: lib/library.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      book if book.author_first_name == name_ary[0]
+    end.compact
```
Comment (BrianZanti) [inline lib/library.rb]: What you were looking for was find_all

**Comment by colehart** [comment]
Thanks, Brian. The find_all was clutch. I’ve refactored the project and all is working now according to your feedback. One question I have about coverage being in the pull request:

I deleted it after the very first push to origin master when I realized it was getting pushed up, then added a .gitignore with coverage in there so it wouldn’t add every time. I’m pretty sure I removed it and git removed the coverage file, too, because when I hopped back on today and checked, it wasn’t on there.

So, my question is: if you mistakenly push coverage and then remove and git rm -r -f it and add coverage to a .gitignore, does it still show up in the pull request?

Thanks,
Cole

**Pull request closed without merging**